php-classes/Emergence/People/Person.php

```diff
@@ -308,6 +308,19 @@ public function validate($deep = true)
         return $this->finishValidation();
     }
 
+    public function setFields($values)
+    {
+        foreach ($values AS $field => $value) {
+
+            // strip any non-digit characters from phone before setting
+            if ($field == 'Phone') {
+                $value = preg_replace('/\D/', '', $value);
+            }
+
+            $this->_setFieldValue($field, $value);
```

Comment:
@jtwiest this operation should be done in an override of `_setFieldValue`, not `setFields` which is just a means to call `_setFieldValue` in bulk but not the only way.

`$User->Phone = $phone` for example would go right into `_setFieldValue` without passing through `setFields`

Your override should check if `$name == 'Phone'` and filter `$value` before passing the work on to and returning the result of `parent::_setFieldValue`

Reply:
@themightychris I actually don't think this is needed anymore, the length update was the big thing but since Phone is a Decimal field it already gets non digets stripped out: 

https://github.com/JarvusInnovations/emergence-skeleton/blob/016b71487da073a0dbabec08c5a47bde00f31b5b/php-classes/ActiveRecord.class.php#L2168